chore: move code to es modules, update @mapeo/schema

tomasciccola opened the pull request:
Since [mapeo-default-config](https://github.com/digidem/mapeo-default-config) depends on this and is using CommonJS modules, this means that we should also update the default-config to use ES modules, which means updating the nodejs version to - at least - v14. 
I seem to remember an issue with updating node on that repo (related to the headless browser to render svgs into pngs) so its important to know if we can safely update that project's node version before moving forward

tomasciccola commented:
I could bother with more refactoring, moving a bunch of stuff to async/await style instead of callback style or we can merge it as-is. @gmaclennan wdyt?

gmaclennan commented on schema/preset.js:
```diff
@@ -1,8 +1,5 @@
-  ...presetSchema,
-  required: presetSchema.required.filter(prop => prop !== 'id')
-}
+export const required = presetSchema.required.filter(prop => prop !== 'id')
```
This can be removed

gmaclennan reviewed (changes requested):
I think we need to keep compat with node12 so that the config builder works, so can you keep the lock file on v1? Do `git checkout main -- package-lock.json` then make sure you are on node v12 then do `npm install`, should revert the lock file to what it was, then update it with the new dep.

ESM works fine in node 12.

tomasciccola commented:
> > I could bother with more refactoring, moving a bunch of stuff to async/await style instead of callback style or we can merge it as-is. @gmaclennan wdyt?
> 
> Let's note that for a follow-up. This should work for now, keeps the PR small and easy to review.

Added [this issue](https://github.com/digidem/id-presets-builder/issues/9) for visibility

gmaclennan commented on lib/utils.js:
```diff
@@ -1,12 +1,13 @@
+import fs from 'node:fs'
+import { promisify } from 'node:util'
+import jsonschema from 'jsonschema'
+const readFile = promisify(fs.readFile)
```
Sorry I missed this the first time round: you can just use https://nodejs.org/docs/latest-v12.x/api/fs.html#fs_fspromises_readfile_path_options (what you have is fine, just sharing in case you didn't know about this)

gmaclennan reviewed (approved):
Looks good to me, but have you tested this on a preset repo to make sure it works? If so then go ahead and merge (unfortunately this repo currently has no tests)

tomasciccola commented:
So, to pass schema validation I'm modifying the objects in place. For all of them we're adding `schemaName` (because it doesn't make sense that configs have this property). For fields we've decided that their `key` field is renamed to `tagKey` in configurations itself (so, no modification in this repo).
But for presets I modified:
* `terms`: in some presets is null so I added `[]` when this the case. We can also avoid this and change presets in configs to have `terms`
* `fieldIds`: it doesn't make sense to have them in configs since this is only useful when imported to the db
* `addTags`: according to the schema, they should default to `tags` so I added that default value if `null`. We can remove the default value here and change presets to have this
* `removeTags`: same than `addTags`, but they default to the value of `addTags` instead of `tags`, so I also added the check. same with `addTags`, we can keep the check, or modify existing presets in configs...

tomasciccola commented:
to add to this, is important to know that we're kinda doing the same thing on config import with [presets](https://github.com/digidem/mapeo-core-next/blob/aad3ce582004a6d1f566f9a8167ca3d53611d21c/src/config-import.js#L161) and [fields](https://github.com/digidem/mapeo-core-next/blob/aad3ce582004a6d1f566f9a8167ca3d53611d21c/src/config-import.js#L116), so I would be inclined to change configurations directly (with good documentation on @mapeo/schema to ease migrations) instead of patching objects in multiple places. @gmaclennan wdyt??

gmaclennan commented:
Hmm it's all a little trickier than we first thought eh? So I think with these fields (terms, addTags, removeTags) we should add the defaults ([] and {}) if the value is undefined or null, as part of this build step, so that the values in the built config do pass the schema validation. That way we can remove the editing of these properties during config import.
